tests/plugins/session.js

```diff
@@ -343,7 +368,29 @@ test('session#routes.login.callback nonce passed', (t) => {
   };
 
   server.inject(options, (response) => {
-    t.equal(response.headers['set-cookie'].length, 2);
+    t.equal(response.headers['set-cookie'].length, 4);
+    const cookies = response.headers['set-cookie'].map(c => c.split(';')[0]);
+    t.equal(cookies[0], 'nonce=');
+    t.equal(cookies[1], 'state=');
+    t.end();
+  });
+});
+
+test('session#routes.login.callback legacy nonce passed', (t) => {
+  const token = { nonce: '123' };
+  const options = {
+    method: 'POST',
+    url: '/login/callback',
+    payload: {
+      id_token: jwt.sign(token, opts.secret)
+    },
+    headers: {
+      Cookie: 'nonce_compat=456; nonce_compat=123;'
```

Comment:
Purely for my own educational purpose, why we are providing the two different `cookie-value`s for the same `cookie-name`? This test case appears to validate that `nonce_compat=123` is accepted while `nounce_compat=456` is ignored. Is this intended to validate certain behavior w.r.t to how the order of the cookie pair is handled?

Reply:
Yes, the intention of this case where two `nonce_compat` cookies are sent is to validate that the second one is the one which is validated against the JWT.  This was borrowed from existing test cases that were in this repo.